front_file_hosting/src/App.vue

```diff
@@ -1,30 +1,88 @@
 <template>
   <div id="nav">
     <router-link to="/">Home</router-link> |
-    <router-link to="/about">About</router-link>
+    <router-link to="/register">Sign up</router-link> |
+    <router-link to="/login">log in</router-link> |
+    <router-link to="/user">User</router-link> |
+    <router-link to="/user/change_password">Change Password</router-link> |
+    <router-link to="/files">Files</router-link> |
+    <router-link to="/upload">Upload</router-link> |
+    <router-link to="/file_detail">File</router-link> |
   </div>
+
+  <div class="container d-flex flex-column flex-md-row align-items-center pb-3 mb-4 border-bottom">
+    <a href="/" class="d-flex align-items-center text-dark text-decoration-none">
+        <span class="fs-4"><i>The best storage for your files</i></span>
+    </a>
+    <nav class="d-inline-flex mt-2 mt-md-0 ms-md-auto">
+        <a class="me-3 py-2 text-dark text-decoration-none" href="#/">Home</a>
+        <a class="me-3 py-2 text-dark text-decoration-none" href="#/register">Sign Up</a>
+        <a class="py-2 me-4 text-dark text-decoration-none" href="#/login">Login</a>
+        <a class="py-2 me-4 text-dark text-decoration-none" href="#/user">Profile</a>
+        <a class="py-2 me-4 text-dark text-decoration-none" href="#/files">Files</a>
+        <a class="btn btn-primary" href="http://127.0.0.1:1338/api/files/template/">Upload File</a>
+    </nav>
+  </div>
+
   <router-view/>
+
 </template>
 
-<style lang="scss">
-#app {
-  font-family: Avenir, Helvetica, Arial, sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-  text-align: center;
-  color: #2c3e50;
-}
+<script>
+import axios from 'axios'
+
+export default {
+    name: 'App',
+    beforeCreate() {
+        this.$store.commit('initializeStore')
+
+        const access = this.$store.state.access
 
-#nav {
-  padding: 30px;
+        if ( access ) {
+            axios.defaults.headers.common['Authorization'] = access
+        } else {
+            axios.defaults.headers.common['Authorization'] = ''
+        }
+    },
+    mounted() {
+        setInterval(() => {
+            this.getAccess()
+        }, 59000)
+    },
+    methods: {
+        getAccess(){
+            const accessData = {
+                refresh: this.$store.state.refresh
+            }
 
-  a {
-    font-weight: bold;
-    color: #2c3e50;
+        axios
+            .post('accounts/refresh/', accessData)
+            .then(response => {
 
-    &.router-link-exact-active {
-      color: #42b983;
+                console.log(response.data.data.result.access_token)
+                console.log(response.data.data.result.refresh_token)
+
+                const access = response.data.data.result.access_token
```

Comment:
you should carefully retrieve data (same as with dicts in python) because you might have errors in case the response shape doesn't fit your structure, I would go with:
```
const responseData = response.data;
if (responseData) {
  const access_token = responseData.data && responseData.data.result && responseData.data.result.access_token
```
there is an optional chaining operator which is cool, but not supported by IE - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining, so you can use it instead of my suggestion